src/app/services/courses/course.service.ts

```diff
@@ -6,45 +6,63 @@ import { Course } from '../../shared/models/course.model';
+            {id: 3, title: 'Video course 3', duration: 135, date: new Date(2017, 12, 17), topRated: false, description: 'In dictum ultricies sed. Massa elit posuere sit scelerisque, nullam commodi, wisi litora interdum tortor. Nulla quam rutrum, mollis odio wisi ac tempus, vehicula wisi tristique eget maecenas, suspendisse et neque elit nonummy arcu congue.'},
+            {id: 4, title: 'Video course 4', duration: 45, date: new Date(2018, 1, 11), topRated: false, description: 'Est penatibus nulla proin eleifend integer nonummy, non consequuntur vestibulum donec nullam penatibus, sociis elementum morbi, consequat ut eget sociis. Integer lacinia ligula blandit vestibulum, commodo volutpat quam quis, pede suspendisse aliquam duis non lectus praesent, proin justo, pede sit sit nec. Ut donec pellentesque et cum, per faucibus ac eu.'}
         ];
+
+        this.courses = Observable.of(courseArray);
     }
 
-    public getAll(): Course[] {
-        return this.courses;
+    public getAll(): Observable<Course[]> {
+        return this.courses
+            .map((courses: Course[]) => courses
+                .filter((x: Course) => {
+                    return x.date.getDate() > (new Date().getDate() - 14);
+                }
+            ));
     }
 
-    public getById(id: number): Course {
-        return this.courses.find((x: Course) => x.id === id);
+    public getById(id: number): Observable<Course> {
+        return this.courses
+            .map((courses: Course[]) => courses.find((course: Course) => course.id === id));
     }
 
-    public create(course: Course): Course {
-        this.courses.push(course);
-        return course;
+    public create(course: Course): Observable<Course> {
+        this.courses
+            .map((courses: Course[]) => courses.push(course));
+
+        return Observable.of(course);
     }
 
-    public update(course: Course): Course {
-        let oldCourse = this.getById(course.id);
-        let index = this.courses.indexOf(oldCourse, 0);
-        if (index > -1) {
-            this.courses.splice(index, 1, course);
-        }
-        return course;
+    public update(newCourse: Course): Observable<Course> {
+        this.courses
+            .map((courses: Course[]) => {
+                let oldCourse = courses.find((course: Course) => course.id === newCourse.id);
+                let index = courses.indexOf(oldCourse, 0);
+                if (index > -1) {
+                    courses.splice(index, 1, newCourse);
+                }
+            });
+
+        return this.getById(newCourse.id);
```

Comment:
the same here. you should not mutate anything within a reactive stream. emit a new portion of data instead